Mimecast/mimecast_modules/connector_mimecast_siem.py

```diff
@@ -0,0 +1,249 @@
+                self.connector.push_events_to_intakes(events=batch_of_events)
+
+            else:
+                self.log(
+                    message=f"{self.log_type}: No events to forward",
+                    level="info",
+                )
+                EVENTS_LAG.labels(intake_key=self.connector.configuration.intake_key).set(0)
+
+        # get the ending time and compute the duration to fetch the events
+        batch_end_time = time.time()
+        batch_duration = int(batch_end_time - batch_start_time)
+        self.log(message=f"{self.log_type}: Fetched and forwarded events in {batch_duration} seconds", level="info")
+        FORWARD_EVENTS_DURATION.labels(intake_key=self.connector.configuration.intake_key).observe(batch_duration)
+
+        # compute the remaining sleeping time. If greater than 0, sleep
+        delta_sleep = self.connector.configuration.frequency - batch_duration
+        if delta_sleep > 0:
+            self.log(message=f"{self.log_type}: Next batch in the future. Waiting {delta_sleep} seconds", level="info")
+            time.sleep(delta_sleep)
+
+    def run(self) -> None:
+        while self.running:
+            try:
+                self.next_batch()
+            except Exception as error:
+                self.log_exception(error, message=f"{self.log_type}: Failed to forward events")
+
+
+class MimecastSIEMConnector(Connector):
+    module: MimecastModule
+    configuration: MimecastSIEMConfiguration
+
+    TYPES_TO_GET = ("process", "journal", "receipt")
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self._stop_event = Event()
+
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.context_lock = Lock()
+
+    def start_consumers(self) -> dict[str, MimecastSIEMWorker]:
+        consumers = {}
+        for consumer_name in self.TYPES_TO_GET:
+            consumers[consumer_name] = MimecastSIEMWorker(connector=self, log_type=consumer_name)
+            consumers[consumer_name].start()
+
+        return consumers
+
+    def supervise_consumers(self, consumers: dict[str, MimecastSIEMWorker]) -> None:
+        for consumer_name, consumer in consumers.items():
+            if consumer is None or (not consumer.is_alive() and consumer.running):
+                self.log(message=f"Restarting `{consumer_name}` consumer", level="info")  # pragma: no cover
+
+                consumers[consumer_name] = MimecastSIEMWorker(connector=self, log_type=consumer_name)
+                consumers[consumer_name].start()
+
+    def stop_consumers(self, consumers: dict[str, MimecastSIEMWorker]):
+        for consumer_name, consumer in consumers.items():
+            if consumer is not None and consumer.is_alive():
+                self.log(message=f"Stopping `{consumer_name}` consumer", level="info")  # pragma: no cover
+                consumer.stop()  # pragma: no cover
```

Comment:
Can we use [workers](https://github.com/SEKOIA-IO/sekoia-automation-sdk/blob/main/sekoia_automation/connector/workers.py) instead of rewriting these methods?
For the `log_type`, we can use [`partial`](https://docs.python.org/3/library/functools.html#functools.partial) to set the `log_type` parameter for the `MimecastSIEMWorker` initiator (that will derivate the class with the `log_type` set).